packages/extension-base/src/background/handlers/Tabs.ts

```diff
@@ -20,27 +19,21 @@ import RequestExtrinsicSign from '../RequestExtrinsicSign';
 import State from './State';
 import { createSubscription, unsubscribe } from './subscriptions';
 
-function getSortedAccounts (accounts: SubjectInfo): SingleAddress[] {
+function transformAccounts (accounts: SubjectInfo, withTypes = false): InjectedAccount[] {
   return Object
     .values(accounts)
     .filter(({ json: { meta: { isHidden } } }) => !isHidden)
-    .sort((a, b) => (a.json.meta.whenCreated || 0) - (b.json.meta.whenCreated || 0));
-}
-
-function transformAccounts (accounts: SubjectInfo): InjectedAccount[] {
-  return getSortedAccounts(accounts)
-    .map(({ json: { address, meta: { genesisHash, name } } }): InjectedAccount => ({
-      address, genesisHash, name
+    .sort((a, b) => (a.json.meta.whenCreated || 0) - (b.json.meta.whenCreated || 0))
+    .map(({ json: { address, meta: { genesisHash, name } }, type }): InjectedAccount => ({
```

Comment:
If !withTypes, we just need to filter ETH, i.e. that is probably a user that will break. (Think of polkaseembly when it starts receiving these, all hell will break loose... :))